File: golang/daily21/m01/d04.go
package y2021m01

var memo = map[int]int{0: 0, 1: 1}

// 每日一题 20210104
//
// 509. 斐波那契数
// Link: https://leetcode-cn.com/problems/fibonacci-number/
func fib(n int) int {
	if v, ok := memo[n]; ok {
		return v
	}
	memo[n] = fib(n-2) + fib(n-1)
	return memo[n]
}

File: typescript/daily21/m07/d04.ts
import { runAllTest } from "../../utils/test"
/**
 * 645. 错误的集合
 * 
 * @link https://leetcode-cn.com/problems/set-mismatch/
 */
function findErrorNums(nums: number[]): number[] {
  const len = nums.length + 1;
  const counter: number[] = new Array(len);
  for (const n of nums) {
    counter[n] = counter[n] ? counter[n] + 1 : 1;
  }
  const ret = [0, len];
  for (let i = 1; i < len; i++) {
    if (counter[i] === 2) {
      ret[0] = i;
    } else if (!counter[i]) {
      ret[1] = i;
    }
  }
  return ret;
};

runAllTest("20210704", findErrorNums, [
  { args: [[1, 2, 2, 4]], res: [2, 3] },
  { args: [[1, 1]], res: [1, 2] },
]);